client/src/Components/PostCreator/PostCreator.js

```diff
@@ -10,28 +10,55 @@ const PostCreator = (props) => {
 	const history = useHistory();
 	const [popup, setPopup] = useState({});
 
-	async function handleClick (e) {
-		e.preventDefault();
-		const formElement = e.currentTarget;
-		const formData = new FormData(formElement);
-		const title = formData.get('title');
-		const body = formData.get('body');
+	async function sendPostRequest (title, body) {
 		const res = await fetch('/api/v1/posts', {
 			'headers': {
 				'Content-Type': 'application/json',
-				'Authorization': `Bearer ${Cookies.get('token')}`
+				'Authorization': `Bearer ${Cookies.get('accessToken')}`
 			},
 			'method': 'POST',
 			'body': JSON.stringify({
 				title,
 				body
 			})
 		});
+		return res;
+	}
+
+	async function sendTokenRequest () {
+		const res = await fetch('/api/v1/tokens/access-token', {
+			'headers': {
+				'Content-Type': 'application/json',
+				'Authorization': `Bearer ${Cookies.get('refreshToken')}`
+			},
+			'method': 'GET'
+		});
+		return res;
+	}
+
+	async function handleClick (e) {
+		e.preventDefault();
+		const formElement = e.currentTarget;
+		const formData = new FormData(formElement);
+		const title = formData.get('title');
+		const body = formData.get('body');
+		let res = await sendPostRequest(title, body);
+		let resend = false;
+		if (res.status === 401) {
+			// try to use the refresh token
+			const refreshRes = await sendTokenRequest();
+			if (refreshRes.status === 200) {
+				resend = true;
+			}
```

Comment:
In this case you'll need to make three sequential requests to make this call work. And moreover, you're going to make this check for all of the requests that require this token. So it isn't so nice approach.
Please, read and try to implement different approach how to handle this expiration for all such requests.

Reply:
Most websites I read just include the refresh token endpoint I already have and show no implementation client-side. Maybe I just can't find one. To fix this I will just attempt to use the client's refresh token if the access token doesn't work on the server